Move types for AWS SDK credentials to a separate crate (#2108)
awslabs/smithy-rs: aws/rust-runtime/aws-credential-types/README.md

# aws-credential-types

> This crate provides the items concerned with AWS SDK credentials including:

Reviewer: Nit: I think "types" is a better word for this
```suggestion
This crate provides types concerned with AWS SDK credentials including:
```

> * A trait for credential providers

Reviewer 2: In this context, we've been using the plural for "credentials" since the provider provides multiple credentials (access key, secret key, and sometimes a token):
```suggestion
* A trait for credentials providers
```

* An opaque struct representing credentials

> * Concrete implementations of credential caching

Reviewer 2: ```suggestion
* Concrete implementations of credentials caching
```
